### PR TITLE
Build Python 3.12

### PR DESCRIPTION
**Pull request by akaszynski**
Update the CI to build Python 3.12.

### CODE REVIEW

**Comment by akaszynski** [comment]
This needs the next release of VTK.

**Comment by RobPasMue** [comment]
VTK 9.3.0 was released yesterday - if that's the case you should be good to go =)

File: .github/workflows/testing-and-deployment.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
```
Comment (RobPasMue) [inline .github/workflows/testing-and-deployment.yml:57]:
```suggestion
    strategy:
      fail-fast: false
      matrix:
```

**Comment by RobPasMue** [comment]
Added a change to allow ``fail-fast: false`` - we should just be able to re-run the failed matrix jobs rather than having to run the entire matrix again. Also, this project is public so the computational resources are "free of charge" =)

File: .ci/build_wheels.sh [hunk 2]
```diff
@@ -5,28 +5,12 @@ set -e -x
...
-  ;;
-esac
+PYBIN="/opt/python/cp${PYTHON_VERSION//.}-cp${PYTHON_VERSION//.}/bin"
 
 # build, don't install
 cd io
 "${PYBIN}/pip" install build
 "${PYBIN}/python" -m build --wheel
-auditwheel repair dist/ansys_mapdl_reader*.whl
+auditwheel repair dist/mapdl_archive*.whl
```
Comment (RobPasMue) [inline .ci/build_wheels.sh:14]:
```suggestion
auditwheel repair dist/ansys_mapdl_reader*.whl
```

The wheel name should be ``ansys_mapdl_reader`` if I am not mistaken. That's the package name.

**Comment by RobPasMue** [comment]
I am really looking forward for the moment they provide a ``retry`` argument for matrix jobs (natively) - especially for those projects in which some of the jobs are flaky, But well... still waiting for GitHub to implement it. There are some actions out there that would do the job but I am not inclined to use them (yet).

**Review by RobPasMue — APPROVED** [review]

**Comment by RobPasMue** [comment]
Looks like the macOS wheels are not being built for Python 3.12 - just saw this

https://github.com/pypa/cibuildwheel/blob/main/action.yml#L31

https://github.com/pypa/cibuildwheel/blob/v2.16.2/action.yml#L31

Looks like the Python versions were never provided as inputs so this is effectively doing nothing... Once the new version of ``pypa/cibuildwheel`` is out we should be able to generate them.

**Pull request merged**